esdt-safe/src/lib.rs

```diff
@@ -274,16 +274,18 @@ pub trait EsdtSafe:
         // This is passed by the BridgeTokenWrapper contract
         let mut is_refund_tx = false;
         let caller = self.blockchain().get_caller();
-        let user_addr = match opt_refund_address {
-            OptionalValue::Some(addr) => {
-                require!(
-                    caller == self.bridged_tokens_wrapper_address().get(),
-                    "Wrong caller for a refund tx"
-                );
-                is_refund_tx = true;
-                addr
+        let user_addr = match opt_refunding_address {
```

Comment:
the logic is this:
allow to set the opt_refunding_address if the caller is either the bridge proxy or wrapper, otherwise, signal error (for example, setting this parameter on a user tx, directly will cause an user error)
Then, when the caller is the bridge proxy, set the is_refund_tx flag to true

Reply:
exactly